presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -67,12 +77,14 @@ public HiveMetastoreClosure(HiveMetastore delegate)
 
     public PartitionStatistics getTableStatistics(HiveIdentity identity, String databaseName, String tableName)
     {
-        return delegate.getTableStatistics(identity, databaseName, tableName);
+        return delegate.getTableStatistics(identity, getExistingTable(identity, databaseName, tableName));
```

Comment:
If I grasp the idea of the closure class correctly, it should have no logic, but should allow lower layer call back and benefit from layers higher up.

I.e. if `delegate.getTableStatistics` can answer the call directly, it should do so (we don't need to pass `Table` object in there)
If it cannot, it should explicitly call back the closure to get the `Table` object

let's discuss

Reply:
> I.e. if delegate.getTableStatistics can answer the call directly, it should do so (we don't need to pass Table object in there)

`getTableStatistics` requires `Table` object. It used to fetch it itself, but it leaked cache then.

> If it cannot, it should explicitly call back the closure to get the Table object

I think it would be less clean as then `HiveMetastore` would accept `HiveMetastore` instances from below and up thus making the hierarchy more complex (e.g how to deal with transactional and global cache?). I think passing simple POJOs it the easiest solution here. We already do that in `getPartition`